packages/material-ui-lab/src/DateRangePicker/DateRangePickerViewMobile.tsx

```diff
@@ -28,7 +28,7 @@ interface DesktopDateRangeCalendarProps<TDate>
   changeMonth: (date: TDate) => void;
 }
 
-const onlyDateView = ['date'] as ['date'];
+const onlyDayView = ['day'] as ['day'];
```

Comment:
Should be

```suggestion
const onlyDayView = ['day'] as const;
```

but most of our typings conflict with readonly arrays. We should change all `T[]` props to `readonly T[]` props at some point to ensure compat with `as const` casts i.e. apply https://github.com/mui-org/material-ui/issues/24639 consistently.